sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UnnecessaryUsings.cs

```diff
@@ -88,13 +88,15 @@ private class CSharpRemovableUsingWalker : CSharpSyntaxWalker
             private readonly IImmutableSet<EquivalentNameSyntax> usingDirectivesFromParent;
             private readonly INamespaceSymbol currentNamespace;
             public readonly HashSet<INamespaceSymbol> necessaryNamespaces;
+            private bool linqQueryVisited;
 
             public CSharpRemovableUsingWalker(SyntaxNodeAnalysisContext context, IImmutableSet<EquivalentNameSyntax> usingDirectives, INamespaceSymbol currentNamespace)
             {
                 this.context = context;
                 this.usingDirectivesFromParent = usingDirectives;
                 this.necessaryNamespaces = new HashSet<INamespaceSymbol>();
                 this.currentNamespace = currentNamespace;
+                this.linqQueryVisited = false;
```

Comment:
It is not necessary but I always prefer to be explicit that simply relying on default primitive values. Updated to the field initialization version to avoid confusion in constructor